mobile/App.tsx

```diff
@@ -1,43 +1,82 @@
-import React, { useRef } from 'react';
+import React, { useEffect, useRef } from 'react';
 import { WebView, WebViewMessageEvent } from 'react-native-webview';
 import { SafeAreaView, Text, Platform } from 'react-native';
 import { torClient } from './services/Tor';
+import Clipboard from '@react-native-clipboard/clipboard';
 
 const App = () => {
   const webViewRef = useRef<WebView>();
   const uri = (Platform.OS === 'android' ? 'file:///android_asset/' : '') + 'Web.bundle/index.html';
 
-  const injectMessage = (id: string, data: object) => {
+  const injectMessageResolve = (id: string, data: object) => {
     const json = JSON.stringify(data);
     webViewRef.current?.injectJavaScript(
       `(function() {window.NativeRobosats.onMessageResolve(${id}, ${json});})();`,
     );
   };
 
+  const injectMessage = (message: object) => {
+    const json = JSON.stringify(message);
+    webViewRef.current?.injectJavaScript(
+      `(function() {window.NativeRobosats?.onMessage(${json});})();`,
+    );
+  };
+
   const onMessage = async (event: WebViewMessageEvent) => {
     const data = JSON.parse(event.nativeEvent.data);
     if (data.category === 'http') {
+      sendTorStatus();
+
+      console.log(data);
       if (data.type === 'get') {
-        torClient.get(data.path).then((response: object) => {
-          injectMessage(data.id, response);
-        });
+        torClient
+          .get(data.path)
+          .then((response: object) => {
+            injectMessageResolve(data.id, response);
+          })
+          .finally(sendTorStatus);
```

Comment:
I tried first to check and send the status of the daemon every second, but it eventually kills the daemon because somehow it's not able to manage to many status requests at the same time. OS I decided to check the status before and after a request is made